andela/Titan-Devs
# #163518683 User can get email verification

#### What does this PR do?
Email verification 
#### Description of Task to be completed?
After a user has been created successfully the platform sends the email via email,
the user is able to click on the link in order to get access to other services like creating an article.
The is an option of resending the email if the email wasn's sent on time.
EndPoints:
* `POST api/v1/users` for creating an account to the system, and receive a verification email.
* `GET api/v1/users/confirm/:auth_token` for email verification.
* `PUT api/v1/users/resend` for resending email verification.
#### How should this be manually tested?
* Fetch the branch `ft-user-get-verification-link-163518683`
* Checkout the branch `ft-user-get-verification-link-163518683`
* Setup your database. 
* Run tests `yarn run test`
* Start the server `yarn run server`
* Create a user `POST: /api/v1/users` with `email, password, and username`
* Go to your email click on the `verify my email` which was sent while registering it will trigger this endPoint `GET api/v1/users/confirm/:auth_token` using Postman client. 
* You should see a User verified successfully. 
* If the email didn't come you could use this endpoint to resend the email `PUT api/v1/users/resend` you only provide email as follow:
`
{
	"email":"example@gmail.com"
}
`
#### What are the relevant pivotal tracker stories?
163518683
